rotkehlchen/chain/ethereum/modules/uniswap/v3/decoder.py

```diff
@@ -57,93 +68,237 @@ def _maybe_decode_v3_swap(  # pylint: disable=no-self-use
-                    )
-                ):
-                    event.event_type = HistoryEventType.TRADE
-                    event.event_subtype = HistoryEventSubType.SPEND
-                    event.counterparty = CPT_UNISWAP_V3
-                    event.notes = f'Swap {event.balance.amount} {crypto_asset.symbol} in uniswap-v3 from {event.location_label}'  # noqa: E501
-                    out_event = event
-                elif (
-                    (
-                        (event.event_type in (HistoryEventType.RECEIVE, HistoryEventType.TRANSFER)) and  # noqa: E501
-                        (event.location_label == buyer or event.asset == A_ETH) and
-                        event.balance.amount == asset_normalized_value(
-                            amount=amount_received,
-                            asset=crypto_asset,
-                        )
-                    )
-                ):
-                    # In this branch of the condition we also add the transfer to correctly decode
-                    # the case where ETH was sent to the user in a multi-step swap that was decoded
-                    # by the event in the previous step. The check for the location label and the
-                    # amount ensure that we are decoding the right event.
-                    event.event_type = HistoryEventType.TRADE
-                    event.event_subtype = HistoryEventSubType.RECEIVE
-                    event.counterparty = CPT_UNISWAP_V3
-                    event.notes = f'Receive {event.balance.amount} {crypto_asset.symbol} in uniswap-v3 from {event.location_label}'  # noqa: E501
-                    in_event = event
-                elif (
-                    event.event_type == HistoryEventType.RECEIVE and
-                    event.balance.amount != asset_normalized_value(
-                        amount=amount_received,
-                        asset=crypto_asset,
-                    ) and event.counterparty == UNISWAP_V3_ROUTER
-                ):
-                    # this is to make sure its the amm issuing the refund and not an agreggator making a swap  # noqa: E501
-                    # Those are assets returned due to a change in the swap price
-                    event.event_type = HistoryEventType.TRANSFER
-                    event.event_subtype = HistoryEventSubType.NONE
-                    event.counterparty = CPT_UNISWAP_V3
-                    event.notes = f'Refund of {event.balance.amount} {crypto_asset.symbol} in uniswap-v3 due to price change'  # noqa: E501
-
-        maybe_reshuffle_events(out_event=out_event, in_event=in_event)
+        if transaction.to_address not in UNISWAP_ROUTERS:
+            return decoded_events  # work only with the known routers for now
+
+        if (decoded_events[1].asset == A_ETH and decoded_events[1].counterparty in UNISWAP_ROUTERS):  # noqa: E501
+            if decoded_events[1].event_type == HistoryEventType.SPEND:
```

Comment:
I would defend against index errors in case something goes wrong and the data you expect here doesn't exists